include/sys/spa.h

```diff
@@ -208,7 +209,7 @@ typedef struct zio_cksum_salt {
  * G		gang block indicator
  * B		byteorder (endianness)
  * D		dedup
- * X		encryption (on version 30, which is not supported)
+ * X		encryption
```

Comment:
I think this should say `set to zero; see diagram below for encrypted blocks`, and then there should be a separate diagram that shows the layout for encrypted blocks, similar to how we explain embedded BP's (IIRC, checksum[2-3], fill, and dva[2] have different meanings when the X bit is set).  The legend of the new diagram might omit fields that are already described here.